lib/hyperion/result_handling/result_maker.rb

```diff
@@ -35,6 +36,8 @@ def make_from_typho(typho_result, result_class)
       read_body = proc { read(typho_result.body, route.response_descriptor) }
       code = typho_result.code
 
+      log_error_response(read(typho_result.body, :json)) unless typho_result.success?
```

Comment:
Refactoring code doesn't change functionality, so make whatever improvements you want. If the tests still pass, you can be confident.

_Changing/adding/removing_ functionality is a different story. I disagree with the notion that any developer should be able to modify part of a component without first becoming familiar with the entire component. That kind of thinking is what lead to ascent-web being such a mess (as of last winter, at least), with the same thing being done 10 different ways, entities being used inconsistently, conflated concerns (e.g., gateways creating UI-specific structures), conflicting naming conventions being used, duplicate code all over the place, dead code being left around, tests not testing what they intended to test. Understanding the design and implementation philosophy of the component enables you to make the _best_ change instead of the _fastest_. Our goal as a development team is not to crank out feature after feature; it is to steadily mature a platform of capabilities that will support customer and business needs both now and in the future.

Of course I'm not saying code quality doesn't matter. Code should be easy to follow _once you understand the context_. Expecting code generally to be understandable outside of context is naive.

One doesn't have to become an expert on a component before making changes, just comfortable with the basic datatypes, execution paths, and design and testing philosophy.